src/main/scala/in/ashwanthkumar/suuchi/rpc/Router.scala

```diff
@@ -0,0 +1,58 @@
+package in.ashwanthkumar.suuchi.rpc
+
+import io.grpc.ServerCall.Listener
+import io.grpc._
+import io.grpc.netty.NettyChannelBuilder
+import io.grpc.stub.ClientCalls
+import org.slf4j.LoggerFactory
+
+/**
+ * Router decides to route the incoming request to right node in the cluster as defined
+ * by the [[RoutingStrategy]].
+ *
+ * @param routingStrategy
+ */
+class Router(routingStrategy: RoutingStrategy) extends ServerInterceptor {
+  private val log = LoggerFactory.getLogger(getClass)
+
+  override def interceptCall[ReqT, RespT](serverCall: ServerCall[ReqT, RespT], headers: Metadata, next: ServerCallHandler[ReqT, RespT]): Listener[ReqT] = {
+    log.debug("Intercepting " + serverCall.getMethodDescriptor.getFullMethodName + " method")
+    new Listener[ReqT] {
+      val delegate = next.startCall(serverCall, headers)
+      var forwarded = false
+
+      override def onReady(): Unit = delegate.onReady()
+      override def onMessage(incomingRequest: ReqT): Unit = {
+        // TODO - Handle forwarding loop here
+        if(routingStrategy.route.isDefinedAt(incomingRequest)) {
```

Comment:
It helps when you don't want to forward certain kind of requests like health / ping. I know we can choose not to add this intercept for them, but I felt this also gives greater power in the hands of the developer. On the other hand it could also be deemed as confusing because of more than 2 ways to do the same thing. What do you think we should be doing?